packages/web-components/@sfx/product/test/unit/product.test.ts

```diff
@@ -0,0 +1,231 @@
+      expect(component.product.price).to.equal(12.34);
+    });
+  });
+
+  describe('additionalInfo', () => {
+    it('should return an empty array if there are no additional properties', () => {
+      const result = component.additionalInfo();
+      expect(result).to.eql([]);
+    });
+
+    it('should return an array of TemplateResults there are additional properties', () => {
+      component.product = {
+        name: '',
+        salePrice: 0,
+        promo: '30% off'
+      };
+
+      const result = component.additionalInfo();
+
+      expect(result[1]).to.be.an.instanceOf(TemplateResult);
+    });
+  });
+
+  describe('urlWrap', () => {
+    it('should return the children wrapped in an "a" tag', () => {
+      const result = component.urlWrap('string', html`<span>foo</span>`);
+      expect(result).to.be.an.instanceof(TemplateResult);
+    });
+
+    it('should return the children', () => {
+      const result = component.urlWrap(undefined, html`<span>foo</span>`);
+      expect(result).to.be.an.instanceof(TemplateResult);
+    });
+  });
+
+  describe('render', () => {
+    it('should return an instance of TemplateResult', () => {
+      const result = component.render();
+      expect(result).to.be.an.instanceof(TemplateResult);
+    });
+
+    it('should call the additionalInfo function', () => {
+      const additionalInfo = stub(component, 'additionalInfo')
+      component.render();
+      expect(additionalInfo).to.be.called;
+    });
+
+    it('should call the urlWrap function', () => {
+      const urlWrap = stub(component, 'urlWrap')
+      component.render();
+      expect(urlWrap).to.be.called;
+    });
+
+  });
+})
+
+describe('Variant Component', () => {
+  let component;
+
+  beforeEach(() => {
+    component = new Variant();
+  });
+
+  describe('constructor', () => {
+    it('should extend the Base class', () => {
+      expect(component).to.be.an.instanceOf(Base);
+    });
+
+    describe('type property', () => {
+      it('should have a default value of "full"', () => {
+        expect(component.type).to.equal('text');
+      });
+    });
+
+    describe('variant property', () => {
+      it('should have a default value of "full"', () => {
+        expect(component.variant).to.eql({ text: '', product: {} });
+      });
+    });
+  });
+
+  describe('connectedCallback', () => {
+    it('should call its super connectedCallback', () => {
+      const connectedCallback = stub(Base.prototype, 'connectedCallback');
+
+      component.connectedCallback();
+      expect(connectedCallback).to.have.been.called;
+    });
+
+    it('should add an eventListener to the window', () => {
+      const addEventListener = spy(component, 'addEventListener');
+
+      component.connectedCallback();
+      expect(addEventListener).to.have.been.called;
+    });
+
+    it('should set role to "listitem"', () => {
+      component.connectedCallback();
+      expect(component.getAttribute('role')).to.equal('listitem');
```

Comment:
This test should be stubbing `setAttribute` and asserting that it was called with the proper arguments instead.